### PR TITLE
EQL: Introduce case insensitive variant in~

### LINKED ISSUE #68172: EQL: Introduce in~ (case-insensitive in)
Introduce `in~` to indicate a case insensitive operation.

For example:
```
process where name in~ ("ExPLorEr.eXe") 
```
will match name against `explorer.exe` regardless of case. 
```
process where name in~ ("explorer*") 
```
will match name against `explorer*` verbatim.

### PR DESCRIPTION
**Pull request by costin**
Complement to in, in~ does case-insensitive matching without any
pattern matching - : should be used instead.

process where name in~ ("ExPLorEr.eXe")

will match name against explorer.exe regardless of case.

Fix #68172

### CODE REVIEW

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/ExpressionBuilder.java [hunk 1]
```diff
@@ -167,8 +168,13 @@ public Expression visitOperatorExpressionDefault(EqlBaseParser.OperatorExpressio
         switch (predicate.kind.getType()) {
             case EqlBaseParser.SEQ:
                 return Predicates.combineOr(expressions(predicate.constant()).stream()
+                    .map(c -> new InsensitiveWildcardEquals(source, expr, c, zoneId))
```
Comment (matriv) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/ExpressionBuilder.java:171]: Is this correct? shouldn't this be for the `IN_INSESITIVE` and for `SEQ` the `InsensitiveEquals`?

Comment (costin) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/ExpressionBuilder.java:171]: No. See the tests - `:` or SEQ supports wildcards while `in~` does not, hence why it is only insensitive.

Comment (matriv) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/ExpressionBuilder.java:171]: Oops, sorry, I confused myself.

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/SqlBaseLexer.java [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-import org.antlr.v4.runtime.Vocabulary;
-import org.antlr.v4.runtime.VocabularyImpl;
-import org.antlr.v4.runtime.atn.ATN;
-import org.antlr.v4.runtime.atn.ATNDeserializer;
-import org.antlr.v4.runtime.atn.LexerATNSimulator;
-import org.antlr.v4.runtime.atn.PredictionContextCache;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.TokenStream;
+import org.antlr.v4.runtime.*;
```
Comment (matriv) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/SqlBaseLexer.java:7]: Please revert star imports.

**Review by matriv — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/SqlBaseLexer.java (hunk 2 above)
Comment (costin) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/SqlBaseLexer.java:7]: That looks like a side-effect on ANTLR generation.

**Review by bpintea — APPROVED** [review]
LGTM

**Review by astefan — APPROVED** [review]
LGTM

**Pull request merged**